src/pages/steps/AgeRangeStep.tsx

```diff
@@ -0,0 +1,239 @@
+import { FormEvent, useEffect, useState } from 'react';
+import { Controller } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { css } from 'styled-components';
+
+import { Alert, AlertVariants } from '@/ui/Alert';
+import { Button, ButtonVariants } from '@/ui/Button';
+import { Inline } from '@/ui/Inline';
+import { Input } from '@/ui/Input';
+import { getStackProps, Stack } from '@/ui/Stack';
+import { Text } from '@/ui/Text';
+import { getValueFromTheme } from '@/ui/theme';
+
+const getValue = getValueFromTheme('ageRange');
+
+const AgeRanges = {
+  ALL: { apiLabel: '-' },
+  TODDLERS: { label: '0-2', apiLabel: '0-2' },
+  PRESCHOOLERS: { label: '3-5', apiLabel: '3-5' },
+  KIDS: { label: '6-11', apiLabel: '6-11' },
+  TEENAGERS: { label: '12-15', apiLabel: '12-15' },
+  YOUNGSTERS: { label: '16-26', apiLabel: '16-26' },
+  ADULTS: { label: '18+', apiLabel: '18-' },
+  SENIORS: { label: '65+', apiLabel: '65-' },
+  CUSTOM: {},
+} as const;
+
+const AgeRangeStep = ({
+  field,
+  control,
+  onChange,
+  ...props
+}: {
+  field: any;
+  control: any;
+  onChange: any;
+}) => {
+  const { t } = useTranslation();
+
+  const [showCustomAgeRange, setShowCustomAgeRange] = useState(false);
```

Comment:
```suggestion
  const [isCustomAgeRangeVisible, setIsCustomAgeRangeVisible] = useState(false);
```

Just a suggestion because `showCustomAgeRange` could be seen as a function